lib/spack/spack/test/conftest.py

```diff
@@ -70,6 +70,25 @@ def clean_user_environment():
         ev.activate(active)
 
 
+#
+# Disable checks on compiler executable existence
+#
+@pytest.fixture(scope='function', autouse=True)
+def mock_compiler_executable_verification(request, monkeypatch):
```

Comment:
Where is this fixture used?

Reply:
It's `autouse=True`, it's used for every test except the three new ones.

There are ~100 tests that fail without it, and I initially started to add it to all of them. But then I realized that new tests shouldn't have to consider whether they will use a compiler that has a fake path -- that's a pretty normal test structure to put fake paths in the compiler. So I made it autouse and wrote a way to disable it (like the clean stage checks and the mock_stage fixture)